typing/typecore.ml

```diff
@@ -4583,6 +4583,10 @@ and type_cases ?in_function env ty_arg ty_res partial_flag loc caselist =
     match caselist with
       [{pc_lhs}] when is_var pc_lhs -> false
     | _ -> true in
+  let partial =
```

Comment:
This hoisting seems fine, but may be confusing, as this is unrelated to the `partial` at the end of the function.

Reply:
Fair point. I renamed this to `take_partial_instance`, that should be enough to avoid any confusion.